warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/FieldHarvester.java

```diff
@@ -0,0 +1,271 @@
+    public static final String RAW_FILE_FIELDNAME = "RAW_FILE";
+    public static final String LOAD_SEQUENCE_FILE_NAME = "ingest.event.mapper.load.seq.filename";
+    public static final String TRIM_SEQUENCE_FILE_NAME = "ingest.event.mapper.trim.sequence.filename";
+    public static final String LOAD_RAW_FILE_NAME = "ingest.event.mapper.load.raw.filename";
+    
+    private boolean createSequenceFileName;
+    private boolean trimSequenceFileName;
+    private boolean createRawFileName;
+    private final DateNormalizer dateNormalizer = new DateNormalizer();
+    
+    private static final String SRC_FILE_DEL = "|";
+    private Exception originalException;
+    
+    public FieldHarvester(Configuration configuration) {
+        this.createSequenceFileName = configuration.getBoolean(LOAD_SEQUENCE_FILE_NAME, true);
+        this.trimSequenceFileName = configuration.getBoolean(TRIM_SEQUENCE_FILE_NAME, true);
+        this.createRawFileName = configuration.getBoolean(LOAD_RAW_FILE_NAME, true);
+    }
+    
+    /**
+     * Updates "fields" with extracted, derived, and automatically generated fields. Will capture
+     *
+     * @param fields
+     *            the Multimap to modify with extracted and generated fields
+     * @param ingestHelper
+     *            interface to use for field extraction
+     * @param value
+     *            the record from which the fields will be extracted
+     * @param offset
+     *            record offset within the source file
+     * @param splitStart
+     *            the splitStart for the record
+     */
+    public void extractFields(Multimap<String,NormalizedContentInterface> fields, IngestHelperInterface ingestHelper, RawRecordContainer value, long offset,
+                    String splitStart) throws Exception {
+        // reset exception-in-extraction tracking
+        this.originalException = null;
+        
+        // "candidateFields" holds the fields that will eventually be added to "fields"
+        Multimap<String,NormalizedContentInterface> candidateFields;
+        
+        try {
+            // parse the record into its candidate field names and values using the IngestHelperInterface.
+            candidateFields = ingestHelper.getEventFields(value);
+        } catch (Exception exception) {
+            // delay throwing the exception to attempt salvaging
+            this.originalException = exception;
+            candidateFields = attemptToSalvageFields(value, ingestHelper);
+        }
+        
+        try {
+            // try adding supplemental fields to candidateFields, whether or not they were salvaged
+            addSupplementalFields(value, offset, splitStart, ingestHelper, candidateFields);
+        } catch (Exception exception) {
+            if (null == this.originalException) {
+                this.originalException = exception;
+            } else {
+                // preserve original exception and log the latest exception
+                log.error(exception);
```

Comment:
Might be helpful here to add a logline that this exception occurred while adding supplemental fields. It'll just add a little bit of context maybe? Instead of just a stacktrace popping up in the log.

```
logger.error("Encountered another exception while adding supplemental fields", exception);
```
Maybe?